### PR TITLE
Make possible to set tabSize

### PR DESCRIPTION
**Pull request by huhgawz**

### CODE REVIEW

**Comment by tonyganch** [comment]
Could you add some tests as examples, please?
Shouldn't `\t` be counted as just one character (since it is, well, just one character)?

**Comment by huhgawz** [comment]
Certainly `\t` is a single character and should be counted like that. However, let me explain you  the use case for which setting the `tabSize` within the `options` of the `parse` API is useful. Basically I'm trying to implement something similar to [node-jscs](https://github.com/jscs-dev/node-jscs) but for CSS files. This is, CSS files are linted based on predefined code style rules. By using this project, your favorite IDE (Sublime Text, WebStorm, etc) could be able to lint your CSS files and tell you exactly where a predefined code style rule is being broken. The problem is that IDEs can be configured to consider `\t` characters as multiple space characters. As you can see in the image below, line 33 is indented using a `\t`, however, the IDE indicates that the cursor is at line 33 column 6. 
Screen Shot 2015-02-02 at 10 59 07 pm
For sure for my project I could calculate the correct column for each token. However this would require extra processes which could be avoided by adding this feature into your code.
If you are ok with this feature I will proceed to generate a couple of tests.

**Comment by tonyganch** [comment]
I see your point. Let me think of it for a day or two?

**Comment by huhgawz** [comment]
Sure. Think about it and let me know once you take a decision.

**Comment by tonyganch** [comment]
Why do you want this option to work only for `.css` files?

**Comment by huhgawz** [comment]
No reason in particular. If you give thumbs up, of course this feature can be added to the rest of the supported formats SCSS, Sass and LESS.

**Comment by huhgawz** [comment]
Any decision?

**Comment by tonyganch** [comment]
@huhgawz, oh, sorry, I've forgotten to answer :) I'm ok with option. Let's add it to all syntaxes and write some tests?

**Comment by huhgawz** [comment]
`tabSize` has been added to all syntaxes. Tests have also been generated.

File: test/tab-size.js [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+ * @param {Object} [options] Parsing options
+ * @return {number} Start column value
+ */
+function getColorDeclarationStartCol(s, options) {
+    var ast = gonzales.parse(s, options);
+    if(options && options.syntax === 'sass') {
+        return ast
+            .first('ruleset')
+            .last('selector')
+            .first('simpleSelector')
```
Comment (tonyganch) [inline test/tab-size.js:24]: Why do we look at selector here if `\t` symbol is right after it?

File: test/tab-size.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+            assert.equal(startCol, 2);
+        });
+    });
+
+    describe('{syntax:\'sass\', tabSize:0}', function() {
+        var options = {syntax: 'sass', tabSize: 0};
+
+        it('should return 5 for tab-indented declaration', function() {
+            var startCol = getColorDeclarationStartCol(sassStringTabIndented, options);
+            assert.equal(startCol, 2);
```
Comment (tonyganch) [inline test/tab-size.js:272]: Two lines above: "should return 5"

File: test/tab-size.js [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        var options = {syntax: 'scss', tabSize: 0};
+
+        it('should return 15 for non-indented declaration', function() {
+            var startCol = getColorDeclarationStartCol(scssStringNonIndented, options);
+            assert.equal(startCol, 15);
+        });
+
+        it('should return 5 for tab-indented declaration', function() {
+            var startCol = getColorDeclarationStartCol(scssStringTabIndented, options);
+            assert.equal(startCol, 2);
```
Comment (tonyganch) [inline test/tab-size.js:328]: Two lines above: "should return 5"

File: lib/parse.js [hunk 4]
```diff
@@ -21,6 +22,7 @@ function parse(css, options) {
     var needInfo = options && options.needInfo || Defaults.NEED_INFO;
     var rule = options && options.rule ||
         (syntax === 'js' ? Defaults.JS_RULE : Defaults.CSS_RULE);
+    var tabSize = options && options.tabSize || Defaults.TAB_SIZE;
```
Comment (tonyganch) [inline lib/parse.js:36]: I think there should be some kind of a check that `tabSize` is a number and is greater than 0.
(Is there a case when you'd want `tabSize` to equal 0?)

File: test/tab-size.js (hunk 1 above)
Comment (huhgawz) [inline test/tab-size.js:24]: The idea is to get the value of the start column of the `color` declaration (`color:$color;`) from the following Sass String:
`var sassStringTabIndented = '$color:blue\na\n\tcolor:$color';`

File: test/tab-size.js (hunk 2 above)
Comment (huhgawz) [inline test/tab-size.js:272]: Take this Saas String as an example:
`var sassStringTabIndented = '$color:blue\na\n\tcolor:$color';`

As you can see, the only indented part is the `color` declaration. It has been indented using a single `\t`.

With regards the `{syntax:'sass'}` test: no `tabSize` option has been set, then `tabSize` option is internally set to 1. Consequently, the value of the `color` declaration start column should be 2 which is correct.

With regards the `{syntax:'sass', tabSize:0}` test: `tabSize` option has been set to 0, then `tabSize` option is internally set to 1. Consequently, the value of the `color` declaration start column should be 2 which is correct.

File: test/tab-size.js (hunk 3 above)
Comment (huhgawz) [inline test/tab-size.js:328]: I have already explained the functionality of the tests in my previous comments.

File: lib/parse.js (hunk 4 above)
Comment (huhgawz) [inline lib/parse.js:36]: Agree. I have added a function that verifies if `tabSize` is an integer greater than 0: 31f885c
I don't think it make sense to set a `tabSize` equal to 0.

**Comment by huhgawz** [comment]
Any update?

**Comment by huhgawz** [comment]
I have been waiting a lot for this code to be merged without response. I feel I have wasted time implementing stuff you required and you never planned to merge.

**Comment by tonyganch** [comment]
@huhgawz, I would like to apologise. This pr was certainly not one of my best moments.
I've merged the changes into `dev` branch, so `tabSize` will be available in next release (I'm planning it somewhere next week).
I'm sorry for not giving you enough attention and for such long silence. I hope to behave better next time.
Thank you for help.

**Pull request closed without merging**